src/portfolios/portfolios.controller.ts

```diff
@@ -254,4 +255,41 @@ export class PortfoliosController {
 
     return handleServiceResult(result);
   }
+
+  @ApiOperation({
+    summary: 'Get list of transactions for a Portfolio',
+    description:
+      'This endpoint will provide list of transaction for the provided Portfolio of an Identity',
+  })
+  @ApiParam({
+    name: 'did',
+    description: 'The DID of the Identity whose Portfolio transactions are to be fetched',
+    type: 'string',
+    example: '0x0600000000000000000000000000000000000000000000000000000000000000',
+  })
+  @ApiParam({
+    name: 'id',
+    description:
+      'The ID of the portfolio for which transactions are to be fetched. Use 0 for the default Portfolio',
+    type: 'string',
+    example: '0x0600000000000000000000000000000000000000000000000000000000000000',
+  })
+  @ApiOkResponse({
+    description: 'Portfolio transactions',
+    type: PortfolioModel,
+  })
+  @Get('/identities/:did/portfolios/:id/transactions')
+  async getTxHistory(
+    @Param() { did, id }: PortfolioDto,
+    @Query() { account, ticker }: GetTransactionsDto
+  ): Promise<ResultsModel<HistoricSettlement>> {
```

Comment:
There should be a `HistoricSettlementModel` that has the information with that. We shouldn't use the SDK return type as the controller return type.

Reply:
Agreed, you will have to create a separate model class for this